src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -124,157 +90,101 @@ public void consume(final String topicName,
-            binaryDecoder, binaryDecoder);
-    return new BinaryConsumerRecord(messageAndMetadata.key(), messageAndMetadata.message(),
-        partitionId, messageAndOffset.offset());
+  private BinaryConsumerRecord createBinaryConsumerRecord(
+    final org.apache.kafka.clients.consumer.ConsumerRecord<byte[], byte[]> consumerRecord,
+    final String topicName,
+    final int partitionId) {
+
+    // KafkaConsumer instances are created with ByteArrayDeserializer so
+    // there is no reason to deserialize record again.
+    return new BinaryConsumerRecord(consumerRecord.key(), consumerRecord.value(),
+        topicName, partitionId, consumerRecord.offset());
   }
 
-  private AvroConsumerRecord createAvroConsumerRecord(final MessageAndOffset messageAndOffset,
-                                                      final String topicName,
-                                                      final int partitionId) {
-    final MessageAndMetadata<Object, Object> messageAndMetadata =
-        new MessageAndMetadata<Object, Object>(topicName, partitionId,
-            messageAndOffset.message(), messageAndOffset.offset(),
-            avroDecoder, avroDecoder);
+  private AvroConsumerRecord createAvroConsumerRecord(
+    final org.apache.kafka.clients.consumer.ConsumerRecord<byte[], byte[]> consumerRecord,
+    final String topicName,
+    final int partitionId) {
+
     return new AvroConsumerRecord(
-        AvroConverter.toJson(messageAndMetadata.key()).json,
-        AvroConverter.toJson(messageAndMetadata.message()).json,
-        partitionId, messageAndOffset.offset());
+        AvroConverter.toJson(avroDeserializer.deserialize(topicName, consumerRecord.key())).json,
+        AvroConverter.toJson(avroDeserializer.deserialize(topicName, consumerRecord.value())).json,
+            topicName,  partitionId, consumerRecord.offset());
   }
 
 
-  private JsonConsumerRecord createJsonConsumerRecord(final MessageAndOffset messageAndOffset,
-                                                      final String topicName,
-                                                      final int partitionId) {
-    final MessageAndMetadata<Object, Object> messageAndMetadata =
-        new MessageAndMetadata<Object, Object>(topicName, partitionId,
-            messageAndOffset.message(), messageAndOffset.offset(),
-            jsonDecoder, jsonDecoder);
-    return new JsonConsumerRecord(messageAndMetadata.key(), messageAndMetadata.message(),
-        partitionId, messageAndOffset.offset());
+  private JsonConsumerRecord createJsonConsumerRecord(
+    final org.apache.kafka.clients.consumer.ConsumerRecord<byte[], byte[]> consumerRecord,
+    final String topicName,
+    final int partitionId) {
+
+    return new JsonConsumerRecord(
+      deserializeJson(consumerRecord.key()),
+      deserializeJson(consumerRecord.value()),
+        topicName, partitionId, consumerRecord.offset());
+  }
+
+  private Object deserializeJson(byte[] data) {
```

Comment:
nitpick: it's a trivial function, but perhaps we can move this to a `JsonConverter` similar to `AvroConverter`?